ngraph/core/src/preprocess/pre_post_process.cpp

```diff
@@ -236,26 +268,47 @@ std::shared_ptr<Function> PrePostProcessor::build(const std::shared_ptr<Function
                 new_param_shape[width_idx] = input->m_tensor_data->get_spatial_width();
             }
         }
-        auto new_param = std::make_shared<op::v0::Parameter>(input->m_tensor_data->get_element_type(), new_param_shape);
-        if (input->m_tensor_data->is_layout_set()) {
-            new_param->set_layout(input->m_tensor_data->get_layout());
-        }
-        // Old param will be removed, so friendly name can be reused
-        new_param->set_friendly_name(param->get_friendly_name());
 
-        // Also reuse names of original tensor
-        new_param->get_output_tensor(0).set_names(param->get_output_tensor(0).get_names());
+        std::vector<std::shared_ptr<ov::Node>> nodes;
+        std::vector<std::shared_ptr<op::v0::Parameter>> new_params;
+
+        // Create separate parameter for each plane. Shape and friendly name is based on color format
+        for (size_t plane = 0; plane < color_info->planes_count(); plane++) {
+            auto plane_shape = color_info->shape(plane, new_param_shape);
+            auto plane_param =
+                std::make_shared<op::v0::Parameter>(input->m_tensor_data->get_element_type(), plane_shape);
+            if (plane < input->m_tensor_data->planes_sub_names().size()) {
+                auto sub_name = std::string("/") + input->m_tensor_data->planes_sub_names()[plane];
```

Comment:
`std::string("/")` is already a part of subname as I see

Reply:
No, it is 'user-defined' names, passed on `set_color_format(NV12_TWO_PLANES, {"MyY", "MyUV"}` these don't use '/'. There is a test verifying that in this case name will be `<param>/MyY' and `<param>/MyUV' in this case